src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -779,7 +779,9 @@ public List<TeamReward> getTeamRewards(Island island) {
 
     @Override
     public void addTeamReward(TeamReward teamReward) {
-        IridiumSkyblock.getInstance().getDatabaseManager().getTeamRewardsTableManager().addEntry(teamReward);
+        CompletableFuture.runAsync(() -> IridiumSkyblock.getInstance().getDatabaseManager().getTeamRewardsTableManager().save(teamReward)).thenRun(() ->
+                IridiumSkyblock.getInstance().getDatabaseManager().getTeamRewardsTableManager().addEntry(teamReward)
+        );
```

Comment:
This saves the record to the database first, which will generate the Id field.

What was happening was the id wasnt set, so all defaulted to 0, and here https://github.com/Iridium-Development/IridiumSkyblock/blob/master/src/main/java/com/iridium/iridiumskyblock/managers/DatabaseManager.java#L85 You can see we are keying by the id.

So when we save, it actually does a save/update and will replace if the key is the same, but since we havent saved to the database yet, the key (Id) isnt set, so is 0, which means that it will override the other values